### ISSUE TITLE
Conflicts resolution handling errors

### ISSUE THREAD

**Issue by Arslan007** [Azure/azure-mobile-apps#809]
## Describe the bug

When conflict occurs for `OfflineTable` on `PushItemsAsync` in `OfflineSqlStore` then error table got an entry for failed item and if you are handling the conflicts manually and push local item again then these are the issues you can face.

1- No matter what offline table you are pushing but`OfflineSqlStore` always sends you `PushFailedException` if there is failed item entry in errors table. For example `Table A` item was failed due to conflicts you got `PushFailedException` and now if you have multiple tables and you are pushing `Table B` and it succeeded on server but `OfflineSqlStore` still sends you `PushFailedException` and item will be linked in `PushResult` from `Table A`.

https://github.com/Azure/azure-mobile-apps/blob/fe3863b759eba07efa329b84257e8a7fa367f532/sdk/dotnet/src/Microsoft.Datasync.Client/Offline/SyncContext.cs#L606

2- If you are handling conflicts manually and there is a failed item entry in errors table and you are pushing item again with correct version and it get succeeded on server but `OfflineSqlStore` will not delete that item from errors table and you will still get the `PushFailedException` because `PushItemsAsync` method checks for errors in the end if there is any failed item entry in errors table it will consider as `PushFailedException`

## To Reproduce

Steps to reproduce the behaviour:

1- Create conflict on your app
2- Kill the app (It will keep failed item in errors table)
3- Relaunch the app
4- Push your all items from offline tables (`Table A`, `Table B`)
5- Or Push corrected item (Local copy with server version)

## Expected behaviour

1- It will nor send `PushFailedException` for Item of `Table A` when pushing the item of `Table B`
2- When failed item is succeeded on remote then it should delete the failed item from errors table that will not cause again `PushFailedException`

## What platforms?

* Client:
  * .Net 7
  * MAUI
  * 6.0.4
  * Windows/iOS

## Suggested change

**Problem 1**
I am not sure this is best solution or not or can be handled better way by just updating `LoadErrorsAsync` to return current table errors. I was handling by filtering errors for first tableNames from the parameters of `PushItemsAsync` as in my case there was always one table being pushed.

```
// ** AP: Filter errors for the current batch table **
errors = errors.Where(error => error.TableName == tableNames.FirstOrDefault()).ToList();

// If the push did not complete successfully, then throw a PushFailedException.
if (batchStatus != PushStatus.Complete || batch.HasErrors(errors))
{
    List<TableOperationError> unhandledErrors = errors.Where(error => !error.Handled).ToList();
    Exception innerException = batch.OtherErrors.Count > 0 ? new AggregateException(batch.OtherErrors) : null;
    await RemoveErrorsAsync(errors.Where(e => e.Status != HttpStatusCode.Conflict && e.Status != HttpStatusCode.PreconditionFailed), cancellationToken).ConfigureAwait(false);
    throw new PushFailedException(new PushCompletionResult(unhandledErrors, batchStatus), innerException);
}
```

**Problem 2**

Again it can be done better way but I am removing the failed item on success in `ExecutePushOperationAsync` method. I had to handle id for common datatypes like `string`, `int` & `Guid`

```
if (removeFromQueueOnSuccess && error == null)
{
    await OperationsQueue.DeleteOperationByIdAsync(operation.Id, operation.Version, cancellationToken).ConfigureAwait(false);
// ** AP: Delete the Errors on Success based on id & TableName **
    var errors = await batch.LoadErrorsAsync(cancellationToken).ConfigureAwait(false);
    await RemoveErrorsAsync(errors
        .Where(e =>
        {
            if (e.TableName != operation.TableName)
            {
                return false;
            }

            var idToken = e.Item["id"];
            if (idToken is JValue jValue)
            {
                var idValue = jValue.Value;
                if (idValue is string stringId)
                {
                    return stringId == (string)result["id"];
                }
                else if (idValue is Guid guidId)
                {
                    return guidId == (Guid)result["id"];
                }
                else if (idValue is int intId)
                {
                    return intId == (int)result["id"];
                }
            }

            return false;
        }), cancellationToken)
        .ConfigureAwait(false);
}
```

https://github.com/Azure/azure-mobile-apps/blob/fe3863b759eba07efa329b84257e8a7fa367f532/sdk/dotnet/src/Microsoft.Datasync.Client/Offline/SyncContext.cs#L891

I can submit a PR if above solution makes sense for these two issues

**Comment by adrianhall** [comment]
I know how to fix this one (I've been eyeing it for a while), so no need for the PR.

**Comment by Arslan007** [comment]
@adrianhall This will handle only problem # 2 and will not handle peoblem# 1. 
For example if I have multiple offline tables and I am pushing on multiple table. If Table1 have conflicts and you are pushing item for Table2. On Successfull push of Table2 it will send this PushFailedException due to failed entry of Table 1 and send Table1 item in PushResult on PushAsync of Table2. Your thoughts ?

**Comment by adrianhall** [comment]
The problem with multiple issues in the same issue report :-)

I'll re-open and fix today.

**Comment by adrianhall** [comment]
I believe 6.0.5 will fix problem #1 as well.  The solution:

1) Update LoadErrorsAsync in the OperationsBatch to load only the tables being pushed.
2) Update the ExecutePushItemsAsync to call LoadErrorsAsync with the list of tables being pushed.
3) Add a backwards compatible LoadErrorsAsync so that the tests continue to work.

I'm not very concerned about increasing test coverage for the new situation right now.  I'm going to fix this in the v8 code base properly.
